ngrep.c

```diff
@@ -105,9 +106,10 @@
  */
 
 uint16_t snaplen = 65535, limitlen = 65535, promisc = 1, to = 100;
-uint16_t match_after = 0, keep_matching = 0, matches = 0, max_matches = 0;
+unsigned long match_after = 0, keep_matching = 0, matches = 0, max_matches = 0;
```

Comment:
IIRC `unsigned long` on most PC hardware will be 32-bit but that is not guaranteed. Instead you should use the C99 fixed width types. In this case either `uint32_t` or `uint64_t`. This will also keep things consistent with the rest of the file.